### ISSUE TITLE
CSS comments inside selectors yields invalid top-level expression

### ISSUE THREAD

**Issue by anlutro** [sass/libsass#941]
The following SCSS (taken from normalize.css) causes an "invalid top-level expression" error.

Removing the `/* 1 */` comment makes it work.

```css
button,
html input[type="button"], /* 1 */
input[type="reset"],
input[type="submit"] {
  -webkit-appearance: button; /* 2 */
  cursor: pointer; /* 3 */
}
```

I can write a sass-spec PR a bit later.

**Comment by telekid** [comment]
This style of commenting is used heavily in a number of common libraries, including [normalize](https://github.com/necolas/normalize.css/) and [nvd3](https://github.com/novus/nvd3).

**Comment by xzyfer** [comment]
Thanks for the report @anlutro. I can confirm this is still an issue. Please pay close attention to the new sass-spec issue format i.e. https://github.com/sass/sass-spec/tree/master/spec/libsass-todo-issues/issue_823

**Comment by mgreter** [comment]
Thanks for the hint @xzyfer. I guess we shold promote the following command set in the future:

```bash
sass --unix-newlines --sourcemap=none -t nested -C "input.scss" "expected_output.css"
sass --unix-newlines --sourcemap=none -t compact -C "input.scss" "expected.compact.css"
sass --unix-newlines --sourcemap=none -t expanded -C "input.scss" "expected.expanded.css"
sass --unix-newlines --sourcemap=none -t compressed -C "input.scss" "expected.compressed.css"
```

**Comment by xzyfer** [comment]
We should bake this into sass-spec cli?

```
sass-spec init --path ....
sass-spec init --issue 123
```

**Comment by mgreter** [comment]
As @am11, myself and others already have put it, we should probaly create a dedicated spec runner inside libsass repo (in C/C++). That way we could include tests for [sourcemaps] [1] and errors/warnings (and other more API consumer specific issues). IMO it's not worth the time to update the current ruby runner to support many more features, since the bindings there are pretty much lacking any real integration with the available API! With 3.2. we can probably provide a plugin to implement something I already proposed in https://github.com/sass/sass-spec/issues/136#issuecomment-66577402. But I like your feature suggestion! I currently have to adjust the path by hand inside perl-libsass spec-runner!

[1]: https://github.com/mgreter/sourcemap.cpp

**Comment by am11** [comment]
Going by this [comparison sheet](http://en.wikipedia.org/wiki/List_of_unit_testing_frameworks#C.2B.2B) on wikipedia, xUnit++ is one of the most feature-complete and modern C++ unit testing framework: https://bitbucket.org/moswald/xunit/wiki/Home. 

I think we can incorporate that one in sassc repo. :)

**Comment by mgreter** [comment]
I currently use [UnitTest++](https://github.com/unittest-cpp) in [sourcemap.cpp](https://github.com/mgreter/sourcemap.cpp).  IMO we could keep the current spec runner and only add an additional native spec runner for API tests. But in the long run a native runner seems the best option!

**Comment by am11** [comment]
That looks pretty solid. :+1:  
It would probably be a good idea to add UnitTest++ as a submodule in sass/libsass/test. Once API tests runner is configured, we can port spec runner stuff in there. It will emit its own executable binary (say `libsass-test`). Eventually, libsass/test may end up with `UnitTest++` and `fixtures/spec` directories (similar to node-sass: https://github.com/sass/node-sass/tree/master/test). :)

**Comment by drewwells** [comment]
A test runner would be awesome for filing bugs :+1:. Currently, it is challenging to properly file issues (with tests) for these types of errors.

**Comment by xzyfer** [comment]
Spec added https://github.com/sass/sass-spec/pull/277

**Comment by xzyfer** [comment]
Although the error condition was fixed in 3.2. the output is still incorrect. The upcoming 3.3. will pass the spec for this issue.

**Comment by mgreter** [comment]
We now have a passing spec test since #1249 was merged :balloon:
